src/screens/Search/AlgoSearchScreen.js

```diff
@@ -0,0 +1,220 @@
+  static navigationOptions = {
+    title: 'Search',
+    headerStyle: {
+      backgroundColor: colors.darkPurple,
+      shadowOpacity: 0,
+      height: 0,
+    },
+    tabBarIcon: ({ tintColor }) => (
+      <Icon ios="ios-search" android="md-search" style={{ fontSize: 24, color: tintColor }} />
+    ),
+  };
+
+  state = {
+    query: undefined,
+    index: 0,
+    routes: [
+      {
+        key: '1',
+        title: 'Anime',
+        apiKey: 'MTc1MWYzYzNiMjVjZjM5OWFiMDc1YWE4NTNkYWMyZjE4NTk2YjkyNjM1YWJkOWIwZTEwM2U1YmUyMjgyODIwY3Jlc3RyaWN0SW5kaWNlcz1wcm9kdWN0aW9uX21lZGlhJmZpbHRlcnM9',
+        indexName: 'production_media',
+      },
+      {
+        key: '2',
+        title: 'Manga',
+        apiKey: 'MTc1MWYzYzNiMjVjZjM5OWFiMDc1YWE4NTNkYWMyZjE4NTk2YjkyNjM1YWJkOWIwZTEwM2U1YmUyMjgyODIwY3Jlc3RyaWN0SW5kaWNlcz1wcm9kdWN0aW9uX21lZGlhJmZpbHRlcnM9',
+        indexName: 'production_media',
+      },
+      {
+        key: '3',
+        title: 'Users',
+        apiKey: 'NGMyZGNjYWU3NDljNzlhZTJjMGFmMmVlNjBlNzE0NTc0OThiMDZmY2ZjNjBkZTc1OWMwNTA1YjAxOTFhMTU5MHJlc3RyaWN0SW5kaWNlcz1wcm9kdWN0aW9uX3VzZXJzJmZpbHRlcnM9',
+        indexName: 'production_users',
+      },
+    ],
+  };
+
+  renderScene = ({ route }) => (
+    <InstantSearch
+      appId={ALGOLIA_ID}
+      apiKey={route.apiKey}
+      indexName={route.indexName}
+      onSearchStateChange={this.handleSearchStateChange}
+    >
+      <SearchBox placeholder={`Search ${route.title}`} />
+      {this.renderSubScene(route)}
+    </InstantSearch>
+  );
+
+  renderSubScene = (route) => {
+    const { query } = this.state;
+    const { navigation } = this.props;
+
+    switch (route.key) {
+      case '3':
+        return (
+          <View>
+            {query && <Hits />}
+          </View>
+        );
+      default: {
+        return (
+          <ScrollView style={styles.scrollView}>
+            {!query &&
+              <TopsList active={route.title.toLowerCase()} mounted navigation={navigation} />}
+            {query && <Hits />}
+          </ScrollView>
+        );
+      }
+    }
+  };
+
+  handleSearchStateChange = (searchState) => {
+    const { query } = searchState;
+    this.setState({ query: query !== '' ? query : undefined });
+  };
+
+  handleIndexChange = index => this.setState({ index });
+
+  renderIndicator = () => <View />;
+
+  renderLabel = ({ route }) => {
+    let labelTextStyle = styles.tabBarText;
+    if (parseInt(route.key) === this.state.index + 1) {
+      labelTextStyle = { ...styles.tabBarText, ...styles.tabBarTextActive };
+    }
+
+    return (
+      <View style={styles.tabBarItem}>
+        <Text style={labelTextStyle}>{route.title}</Text>
+      </View>
+    );
+  };
+
+  renderHeader = props => (
+    <TabBar
+      {...props}
+      style={styles.tabBar}
+      renderIndicator={this.renderIndicator}
+      renderLabel={this.renderLabel}
+    />
+  );
+
+  render() {
+    return (
+      <Container style={styles.container}>
+        <TabViewAnimated
```

Comment:
hmm -- are we replacing ScrollableTabView with this? I should probably update the UserLibrary screen and remove it if so...having multiple libraries that do the same thing is :wutface: